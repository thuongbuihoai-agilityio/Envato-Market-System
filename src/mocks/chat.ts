// Constants
import { AVATAR_POSITION, IMAGES } from '@app/constants';
import { MESSAGE_TIME } from '.';

// TODO: Update later
export const USER_CHATS = [
  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hi, I need more information',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 1000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 3000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 6000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Hi, i am here for help',
    avatarPosition: AVATAR_POSITION.AFTER,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 2000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 4000,
  },
  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 7000,
  },
];

export const MEMBER_CHATS = [
  { uid: 'admin', name: 'Admin' },
  { uid: 'user1', name: 'User 1' },
];
