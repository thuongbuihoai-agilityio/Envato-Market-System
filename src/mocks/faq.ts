export const FAQ_DATA = [
  {
    id: 1,
    question: 'What is the most important part of a dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 2,
    question: 'What are the three types of dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 3,
    question: 'What are examples of admin?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 4,
    question: 'What are 5 benefits of dashboards?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 5,
    question: 'What makes a good dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 6,
    question: 'What are the 3 basic skills of administrator?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
];
