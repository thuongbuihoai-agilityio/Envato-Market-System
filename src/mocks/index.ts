export * from './spending';
export * from './select';
export * from './table';
export * from './sidebar';
