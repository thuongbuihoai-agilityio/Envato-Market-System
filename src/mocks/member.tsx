import { Image } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Utils
import { MESSAGE_TIME_FORMAT } from '.';

export const MEMBER_CHAT = [
  {
    id: 'user1',
    avatar: 'https://example.com/avatar1.jpg',
    name: 'Kien Nguyen',
    status: 'Online',
    localeTime: MESSAGE_TIME_FORMAT,
    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'online',
  },
  {
    id: 'user2',
    avatar: 'https://example.com/avatar2.jpg',
    name: 'Huy Pham',
    status: 'Away',
    localeTime: MESSAGE_TIME_FORMAT,

    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'offline',
  },
];
