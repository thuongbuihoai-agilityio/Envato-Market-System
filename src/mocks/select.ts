export const SOCIAL_PLATFORM_OPTIONS = [
  {
    value: 'github',
    label: 'github',
  },
  {
    value: 'facebook',
    label: 'facebook',
  },
  {
    value: 'instagram',
    label: 'instagram',
  },
];
