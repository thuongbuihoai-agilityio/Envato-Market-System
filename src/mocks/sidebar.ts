// constants
import {
  HELP_ITEM_LIST,
  MENU_ITEM_LIST,
  OTHER_ITEM_LIST,
} from '@app/constants/sidebar';

export const EXPAND_SIDEBAR_MENU_LIST = [
  {
    id: 1,
    title: 'Menu',
    listItem: MENU_ITEM_LIST,
  },
  {
    id: 2,
    title: 'Help',
    listItem: HELP_ITEM_LIST,
  },
  {
    id: 3,
    title: 'Others',
    listItem: OTHER_ITEM_LIST,
  },
];
