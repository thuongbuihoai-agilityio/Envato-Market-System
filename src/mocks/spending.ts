export const TOTAL_EARNINGS_MOCK = {
  title: 'Total earnings',
  total: 7.245,
  growth: 3.5,
  weeklyIncome: [30, 40, 45, 50, 49, 60, 91],
};
