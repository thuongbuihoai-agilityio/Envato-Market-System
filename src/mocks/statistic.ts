export const STATISTICAL = [
  {
    _id: '656ee2dfebaa8bd1cdee2d0a',
    title: 'Total Earnings',
    total: 7245,
    growth: 3.5,
    weeklyIncome: [30, 40, 45, 50, 49, 60, 91],
  },
  {
    _id: '656ee3a2ebaa8bd1cdee2d0b',
    title: 'Total Spending',
    total: 7245,
    growth: 3.5,
    weeklyIncome: [20, 40, 60, 80, 90, 110, 130],
  },
  {
    _id: '656ee413ebaa8bd1cdee2d0c',
    title: 'Spending Goal',
    total: 7245,
    growth: 3.5,
    weeklyIncome: [30, 40, 45, 50, 49, 60, 91],
  },
];
