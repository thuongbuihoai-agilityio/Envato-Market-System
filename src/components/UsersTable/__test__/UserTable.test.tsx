import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserTable from '..';
import { USERS_MOCK } from '@app/mocks';

describe('UserTable component', () => {
  it('renders correctly', () => {
    const { container } = render(<UserTable users={USERS_MOCK} />);
    expect(container).toMatchSnapshot();
  });

  it('renders children when not loading or in error state', () => {
    render(<UserTable users={USERS_MOCK} />);
    expect(screen.getByText('Abdur Rohman')).toBeInTheDocument();
  });
});
