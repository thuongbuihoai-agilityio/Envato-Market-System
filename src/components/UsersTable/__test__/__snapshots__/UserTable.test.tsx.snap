// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable component renders correctly 1`] = `
<div>
  <div
    class="chakra-table__container css-zipzvv"
  >
    <table
      class="chakra-table css-0"
    >
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
        />
      </thead>
      <tbody
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <td
            class="css-i3u192"
          >
            <div
              class="css-1h2sv0j"
            >
              <img
                alt="Image of Abdur Rohman"
                class="chakra-image css-1rqxmuh"
                src="https://cdn-icons-png.flaticon.com/512/5556/5556468.png"
              />
              <div
                class="css-1rr4qq7"
              >
                <h4
                  class="chakra-text css-197ngd6"
                  title="Abdur Rohman"
                >
                  Abdur Rohman
                </h4>
                <p
                  class="chakra-text css-vxhtcb"
                >
                  <span
                    class="chakra-text css-9feaud"
                    title="Finance managers"
                  >
                    Finance managers
                    ⋅
                  </span>
                  <span
                    class="chakra-text css-1rgac5r"
                    title="Jakarta, Indonesia 2 days ago"
                  >
                    Jakarta, Indonesia
                    ⋅
                    2 days ago
                  </span>
                </p>
              </div>
            </div>
          </td>
          <td
            class="css-dq40i2"
          >
            <span
              class="chakra-badge css-3jdwwr"
              title="Full Time"
            >
              Full Time
            </span>
          </td>
          <td
            class="css-dq40i2"
          >
            <span
              class="chakra-badge css-3jdwwr"
              title="Senior Level"
            >
              Senior Level
            </span>
          </td>
          <td
            class="css-1t7ec6l"
          >
            <button
              aria-label="This is the icon action"
              class="chakra-button css-1x5hsk6"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="4"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM1 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM15 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0Z"
                  stroke="#A0AEC0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-i3u192"
          >
            <div
              class="css-1h2sv0j"
            >
              <img
                alt="Image of Devon Rohman"
                class="chakra-image css-1rqxmuh"
                src="https://cdn-icons-png.flaticon.com/512/5556/5556468.png"
              />
              <div
                class="css-1rr4qq7"
              >
                <h4
                  class="chakra-text css-197ngd6"
                  title="Devon Rohman"
                >
                  Devon Rohman
                </h4>
                <p
                  class="chakra-text css-vxhtcb"
                >
                  <span
                    class="chakra-text css-9feaud"
                    title="Finance managers"
                  >
                    Finance managers
                    ⋅
                  </span>
                  <span
                    class="chakra-text css-1rgac5r"
                    title="Jakarta, Indonesia 4 days ago"
                  >
                    Jakarta, Indonesia
                    ⋅
                    4 days ago
                  </span>
                </p>
              </div>
            </div>
          </td>
          <td
            class="css-dq40i2"
          >
            <span
              class="chakra-badge css-3jdwwr"
              title="Full Time"
            >
              Full Time
            </span>
          </td>
          <td
            class="css-dq40i2"
          >
            <span
              class="chakra-badge css-3jdwwr"
              title="Senior Level"
            >
              Senior Level
            </span>
          </td>
          <td
            class="css-1t7ec6l"
          >
            <button
              aria-label="This is the icon action"
              class="chakra-button css-1x5hsk6"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="4"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM1 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0ZM15 2a1 1 0 1 0 2 0 1 1 0 0 0-2 0Z"
                  stroke="#A0AEC0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
